internal/logger/logger.go

```diff
@@ -52,3 +52,10 @@ func GetZapLevelFromEnv() zapcore.Level {
 		return zapcore.InfoLevel // fallback
 	}
 }
+
+// SyncLogger ensures the logger is properly synced
+func SyncLogger() {
```

Comment:
Why do we need a separate function here? Is this being reused heavily?

Reply:
When I initially improved the logger configuration with better formatting, I noticed that some log messages might not be flushed to the output immediately. That's why I added SyncLogger() to ensure all buffered log messages were flushed before the application exits or before critical operations. I was expecting to use the function but its only being called once so removed it.

Reply:
Ok, who is calling this? How is it set up? Also, should logging changes be part of the metrics PR?